src/Microsoft.ML.Vision/ImageClassificationTrainer.cs

```diff
@@ -1450,7 +1450,7 @@ private protected override void SaveCore(ModelSaveContext ctx)
             var buffer = _session.graph.ToGraphDef(status);
             ctx.SaveBinaryStream("TFModel", w =>
             {
-                w.WriteByteArray(buffer.MemoryBlock.ToArray());
+                w.WriteByteArray(buffer.DangerousMemoryBlock.ToArray());
```

Comment:
⚠️ This use of `buffer` need to be protected. The easiest way is to change `var buffer` to `using var buffer`.

Reply:
#Resolved